### PR TITLE
PDB reading

### PR DESCRIPTION
**Pull request by kain88-de**
should be the last fix for #396

### CODE REVIEW

**Comment by richardjgowers** [comment]
If you want to go with default occupancy being 1.0 then change my passes above, but it might make more sense to have an empty list and put some sort of failure option into `get_occupancy`

**Comment by kain88-de** [comment]
I want to opt for a default occupancy of 1 and test that the reader is actually doing that. But I noticed that I can't even access the occupancy's from an `AtomGroup` object. Does it even make sense to read this stuff if the standard object to interact with a trajectory doesn't support the field?

**Comment by richardjgowers** [comment]
I think what needs doing is occupancy needs moving into the Timestep (`.ts` attribute).  Data shouldn't be living in the Reader objects.  So something like
``` python
# Random data just lives in a dict called data
self.ts.data['occupancy'] = np.array(occupancies)
```

Then you can add links to this from both `Atom` and `AtomGroup`

``` python
class Atom
    @property
    def occupancy(self):
        try:
            return self.universe.coord.data['occupancy'][self.index]
        except KeyError:
            raise NoDataError("with nice error message, see velocity for example")
```

```

**Comment by kain88-de** [comment]
I added the occupancy field. But setting is only allowed at the atom level. Setting things at the AtomGroup Level is a little to complex for me (the code). I'll have to look into the correct things for caching and attribute setting.

All of the new code is covered by tests as well.

File: package/MDAnalysis/core/AtomGroup.py [hunk 1]
```diff
@@ -606,6 +606,29 @@ def velocity(self, vals):
...
+        try:
+            return self.universe.coord.data['occupancy'][self.index]
+        except KeyError:
+            raise NoDataError('Timestep does not contain occupancy')
+
+    @occupancy.setter
+    def occupancy(self, _occupancy):
+        """Set occupancy for atom"""
+        try:
+            self.universe.coord.data['occupancy'][self.index] = _occupancy
```
Comment (richardjgowers) [inline package/MDAnalysis/core/AtomGroup.py]: One option here is to allocate the array on a KeyError

File: package/MDAnalysis/core/AtomGroup.py [hunk 2]
```diff
@@ -1129,6 +1152,18 @@ def total_mass(self):
...
+    def occupancy(self):
+        """Access occupancies of atoms
+
+        If available can have a value between 0 and 1
+
+        :Returns: ndarray containing occupancies
+
+        .. versionadded:: 0.12.0
+        """
+        return np.array([atom.occupancy for atom in self._atoms])
```
Comment (richardjgowers) [inline package/MDAnalysis/core/AtomGroup.py]: A better way here is `universe.coord.data['occupancy'][self.indices]`

**Comment by richardjgowers** [comment]
Just add a note in the CHANGELOG and we can merge this

File: package/MDAnalysis/core/AtomGroup.py (hunk 1 above)
Comment (kain88-de) [inline package/MDAnalysis/core/AtomGroup.py]: Just to make sure. The `TimeStep` object number of atoms always refers to all the numbers in the frame independent of the current selection?

Comment (richardjgowers) [inline package/MDAnalysis/core/AtomGroup.py]: Yeah `Timestep` arrays are the master arrays, always len(n_atoms), and index/indices are always global

**Comment by kain88-de** [comment]
I added the CHANGELOG and added the changes you suggested.

**Pull request merged**